File: 백준/Gold/1484. 다이어트/다이어트.py
g = int(input())
x, y = 2, 1 # x: 현재 몸무게 y: 과거 몸무게
result = []

while x <= 100000:
    diff = x * x - y * y
    if diff == g:
        result.append(x)
        x += 1
    elif diff < g:
        x += 1
    else:
        y += 1

if result:
    print('\n'.join(map(str, result)))
else:
    print(-1)

File: 백준/Gold/1484. 다이어트/다이어트.js
const input = require('fs')
	.readFileSync('/dev/stdin', 'utf-8')
	.trim()
	.split(' ');

const g = Number(input[0]);

let x = 2;
let y = 1;
const result = [];

while (x <= 100000) {
	const diff = x * x - y * y;

	if (diff === g) {
		result.push(x);
		x++;
	} else if (diff < g) {
		x++;
	} else {
		y++;
	}
}

console.log(result.length ? result.join('\n') : -1);
